presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -848,6 +848,63 @@ public void testRow()
         assertQuery("SELECT cast(row(nationkey, regionkey) as row(a bigint, b bigint)) FROM nation");
     }
 
+    @Test
+    public void testDecimalRangeFilters()
+    {
+        // Actual session is for the native query runner.
+        Session actualSession = Session.builder(getSession())
+                .setCatalog("hive")
+                .setCatalogSessionProperty("hive", "parquet_pushdown_filter_enabled", "true")
+                .build();
+
+        // Expected session is for the Java query runner.
+        // It does not support Parquet filter pushdown yet, so we have to explicitly disable it.
+        Session expectedSession = Session.builder(actualSession)
+                .setCatalogSessionProperty("hive", "parquet_pushdown_filter_enabled", "false")
+                .build();
+
+        // Generate temporary table name.
+        String tmpTableName = generateRandomTableName();
+
+        try {
+            // Create a Parquet table with decimal types and test data.
+            getExpectedQueryRunner().execute(expectedSession, String.format("CREATE TABLE %s (c0 DECIMAL(15,2), c1 DECIMAL(38,2)) WITH (format = 'PARQUET')", tmpTableName));
+            getExpectedQueryRunner().execute(expectedSession, String.format("INSERT INTO %s VALUES (DECIMAL '0', DECIMAL '0')", tmpTableName));
```

Comment:
Please combine all the insert statements into one single statement with multiple values for conciseness.

Reply:
Updated!